### PR TITLE
Integrate GasPrice component with transfer

### PR DESCRIPTION
**Pull request by xitronix**

### CODE REVIEW

File: universal-login-commons/lib/core/models/transactions.ts [hunk 1]
```diff
@@ -1,9 +1,11 @@
 import {utils} from 'ethers';
+import {GasParameters} from './gas';
 
 export interface TransferDetails {
   to: string;
   amount: string;
-  currency: string;
+  transferToken: string;
```
Comment (marekkirejczyk) [inline universal-login-commons/lib/core/models/transactions.ts]: `transferToken`  -> `gasToken`?

File: universal-login-commons/lib/core/utils/sanitizers/asBigNumber.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import {utils} from 'ethers';
+import {Sanitizer, Result} from '@restless/sanitizers';
+
+export const asBigNumber: Sanitizer<utils.BigNumber> = (value, path) => {
```
Comment (marekkirejczyk) [inline universal-login-commons/lib/core/utils/sanitizers/asBigNumber.ts]: Already exist in restless:
https://github.com/EthWorks/restless/tree/master/ethereum#asbignumber

File: universal-login-commons/lib/core/utils/sanitizers/asGasParameters.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import {asObject, Sanitizer, asString} from '@restless/sanitizers';
+import {GasParameters} from '../../models/gas';
+import {asBigNumber} from './asBigNumber';
+
+export const asGasParameters: Sanitizer<GasParameters> = asObject<GasParameters>({
+  gasToken: asString,
+  gasPrice: asBigNumber
+});
```
Comment (marekkirejczyk) [inline universal-login-commons/lib/core/utils/sanitizers/asGasParameters.ts]: 👍

File: universal-login-commons/test/core/utils/sanitizers.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import {expect} from 'chai';
+import {utils} from 'ethers';
+import {ETHER_NATIVE_TOKEN} from '../../../lib/core/constants/constants';
+import {cast} from '@restless/sanitizers';
+import {asGasParameters} from '../../../lib/core/utils/sanitizers/asGasParameters';
+import {asTransferDetails} from '../../../lib';
+import {asBigNumber} from '../../../lib/core/utils/sanitizers/asBigNumber';
+
+describe('sanitizers', () => {
+  describe('asBigNumber', () => {
```
Comment (marekkirejczyk) [inline universal-login-commons/test/core/utils/sanitizers.ts]: use restless version :)

File: universal-login-react/src/ui/Transfer/Amount/ModalTransferAmount.tsx [hunk 5]
```diff
@@ -17,7 +17,7 @@ export const ModalTransferAmount = ({
   deployedWallet,
   onSelectRecipientClick,
   updateTransferDetailsWith,
-  currency,
+  token,
```
Comment (marekkirejczyk) [inline universal-login-react/src/ui/Transfer/Amount/ModalTransferAmount.tsx]: perhaps `tokenDetails`?

File: universal-login-commons/lib/core/utils/sanitizers/asBigNumber.ts (hunk 2 above)
Comment (xitronix) [reply universal-login-commons/lib/core/utils/sanitizers/asBigNumber.ts]: It doesn't work. I will create PR for restless in the weekend

File: universal-login-react/src/ui/Transfer/Recipient/TransferRecipient.tsx [hunk 6]
```diff
@@ -5,13 +5,14 @@ import './../../styles/transferRecipientDefaults.css';
...
 export interface TransferRecipientProps {
+  symbol: string;
   onRecipientChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
   onSendClick: () => Promise<void>;
   transferDetails: Partial<TransferDetails>;
   transferRecipientClassName?: string;
 }
 
-export const TransferRecipient = ({onRecipientChange, onSendClick, transferRecipientClassName, transferDetails: {amount, currency, to}}: TransferRecipientProps) => {
+export const TransferRecipient = ({onRecipientChange, onSendClick, transferRecipientClassName, transferDetails: {amount, to}, symbol}: TransferRecipientProps) => {
```
Comment (marekkirejczyk) [inline universal-login-react/src/ui/Transfer/Recipient/TransferRecipient.tsx]: symbol is now part of TransferDetails?

File: universal-login-react/src/ui/Transfer/Amount/ModalTransferAmount.tsx (hunk 5 above)
Comment (xitronix) [reply universal-login-react/src/ui/Transfer/Amount/ModalTransferAmount.tsx]: This or next pr?

File: universal-login-react/src/ui/UFlow/UDashboard.tsx [hunk 7]
```diff
@@ -31,7 +32,8 @@ function sanitizeTransferDetails(details: Partial<TransferDetails>) {
 
 export const UDashboard = ({deployedWallet}: UDashboardProps) => {
   const {contractAddress, privateKey, sdk} = deployedWallet;
-  const [transferDetails, setTransferDetails] = useState<Partial<TransferDetails>>({currency: sdk.tokensDetailsStore.tokensDetails[0].symbol});
+  const [transferDetails, setTransferDetails] = useState<Partial<TransferDetails>>({transferToken: sdk.tokensDetailsStore.tokensDetails[0].address} as TransferDetails);
```
Comment (marekkirejczyk) [inline universal-login-react/src/ui/UFlow/UDashboard.tsx]: Extract `InitialTransferDetails`

File: universal-login-react/src/ui/Transfer/Recipient/TransferRecipient.tsx (hunk 6 above)
Comment (xitronix) [reply universal-login-react/src/ui/Transfer/Recipient/TransferRecipient.tsx]: TransferDetails operate on tokenAddresses now

File: universal-login-wallet/src/ui/react/Modals/Transfer/ModalTransfer.tsx [hunk 8]
```diff
@@ -10,8 +10,8 @@ const ModalTransfer = () => {
   const [modal, setModal] = useState('transferAmount');
 
   const {walletService, sdk} = useServices();
-  const [transferDetails, setTransferDetails] = useState({currency: sdk.tokensDetailsStore.tokensDetails[0].symbol} as TransferDetails);
-
+  const [transferDetails, setTransferDetails] = useState({transferToken: sdk.tokensDetailsStore.tokensDetails[0].address} as TransferDetails);
```
Comment (marekkirejczyk) [inline universal-login-wallet/src/ui/react/Modals/Transfer/ModalTransfer.tsx]: extract INTITIAL_TRANSFER_DETAILS?

Comment (xitronix) [reply universal-login-wallet/src/ui/react/Modals/Transfer/ModalTransfer.tsx]: Now it takes the first element from tokensDetailsStore. Do you want to do some hardcoded value?

**Pull request merged**